### PR TITLE
TypeScript and C++ CI using GitHub Actions

### LINKED ISSUE #397: Move to GitHub Actions?
Just wondering if there is any benefit in moving from Travis-CI to GitHub Actions. It seems that GitHub Actions makes it easier to test in Mac and Windows. Example:

* https://github.com/eslint/eslint/blob/master/.github/workflows/ci.yml

### PR DESCRIPTION
**Pull request by nazar-pc**
You can see what it looks like in my fork: https://github.com/nazar-pc/mediasoup/actions

Fixes #397

### CODE REVIEW

**Comment by ibc** [comment]
WOWOWOWOW

@jmillan please take a look to this!

@nazar-pc may you please also remove `.travis-ci` file and edit the corresponding link in the `README.md`?

**Comment by nazar-pc** [comment]
Done

**Comment by jmillan** [comment]
Amazing @nazar-pc !!

**Comment by jmillan** [comment]
@nazar-pc , can we add different GCC & Clang versions in the GH actions as we had in Travis?

**Comment by nazar-pc** [comment]
Added Ubuntu 16.04 and 18.04 alongside clang compiler for all operating systems.

Didn't bother with installing specific GCC/Clang versions, feel free to mess with it if you have time. Here is the description if what GitHub Actions have installed by default: https://github.com/actions/virtual-environments/tree/main/images

**Comment by nazar-pc** [comment]
Here is what the build summary looks like with all of those enabled: https://github.com/nazar-pc/mediasoup/actions/runs/639508661

File: .github/workflows/cpp.yaml [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+            cc: gcc
+            cxx: g++
+          - os: macos-10.15
+            cc: clang
+            cxx: clang++
+          # macOS 11 is in private preview currently
+          #- os: macos-11.0
+          # Windows needs to be tuned before it can run, I didn't bother, feel free to contribute
+          #- os: windows-2019
+        # Just 1 Node.js version should be fine for C++
```
Comment (ibc) [inline .github/workflows/cpp.yaml:41]: Actually in package.json we say that mediasoup is supposed to run in `"node": ">=10"`. Can we also test node 10 and 12 here?

Comment (nazar-pc) [reply .github/workflows/cpp.yaml:41]: This is just for worker builds, which do not depend on Node.js really. TypeScript workflow file is being tested on Node.js 10, 12, 14 and 15

Comment (ibc) [reply .github/workflows/cpp.yaml:41]: absolutely, sorry

**Comment by ibc** [comment]
MERGING! THANKS!

**Comment by jmillan** [comment]
Awesome @nazar-pc!

**Pull request merged**